File: Solutions/estimate_pi.py
# Estimate pi, given that you have random (0,1)

import random
import math

def estimatePi(n: int):

  numOfPoints_Circle: int = 0
  numOfPoints_Total: int = 0

  for _ in range(n):
    x: float = random.uniform(0,1)
    y: float = random.uniform(0,1)
    
    distance: float = math.sqrt(x**2 + y**2)
    
    if distance <= 1:
      numOfPoints_Circle += 1
      
    numOfPoints_Total += 1
    
  estimate: float = 4 * (numOfPoints_Circle / numOfPoints_Total)
    
  return estimate
  
# main
estimatedPi: float = estimatePi(10000000)
print(estimatedPi)

File: Solutions/estimate_pi.js
// Estimate pi, given that you have random (0,1)

function estimatePi(n){
    let numOfPoints_Circle = 0;
    let numOfPoints_Total = 0;

    for (i = 0; i < n; i++){
        let x = Math.random();
        let y = Math.random();

        let distance = Math.sqrt((x*x) + (y*y));

        if (distance <= 1){
            numOfPoints_Circle++;
        }
        numOfPoints_Total++
    }

    let estimate = 4 * (numOfPoints_Circle / numOfPoints_Total);

    return estimate
}


let estimatedPi = estimatePi(10000000);
console.log(estimatedPi);
